### PR TITLE
log region for every command and improve help message

### PR DESCRIPTION
**Pull request by aclevername**
### Description
Closes https://github.com/weaveworks/eksctl/issues/2883

Moved the logging into `NewCtl` which every command that actually uses the region calls. Added a `Silent` field for the edge-cases where this func gets called multiple times, e.g. `get clusters --all-regions`

Biggest concern was ensuring that when we print json/yaml output via `-o json/yaml` that we don't break the stdout. Please check thet `ctl/get` diffs carefully :smile:

### CODE REVIEW

File: pkg/ctl/get/addon.go [hunk 1]
```diff
@@ -47,19 +47,16 @@ func getAddonCmd(cmd *cmdutils.Cmd) {
 }
 
 func getAddon(cmd *cmdutils.Cmd, params *getCmdParams) error {
+	if params.output != printers.TableType {
+		//log warnings and errors to stdout
+		logger.Writer = os.Stderr
+	}
```
Comment (aclevername) [inline pkg/ctl/get/addon.go:53]: this pattern seems to be used a lot in the `get` commands for when we want `json/yaml` output, basically ensures `get addons | jq .` still works, but the user still sees our helpful log messages on `stderr`

File: pkg/ctl/get/labels.go [hunk 2]
```diff
@@ -51,7 +51,6 @@ func getLabels(cmd *cmdutils.Cmd, nodeGroupName string) error {
 	if err != nil {
 		return err
 	}
-	cmdutils.LogRegionAndVersionInfo(cmd.ClusterConfig.Metadata)
```
Comment (aclevername) [inline pkg/ctl/get/labels.go:54]: get labels doesn't have json/yaml support

File: pkg/ctl/utils/describe_stacks.go [hunk 3]
```diff
@@ -66,16 +66,16 @@ func doDescribeStacksCmd(cmd *cmdutils.Cmd, all, events, trail bool, printer pri
 		return err
 	}
 
-	cfg := cmd.ClusterConfig
+	if printer != nil {
+		cmdutils.Silent = true
+	}
```
Comment (aclevername) [inline pkg/ctl/utils/describe_stacks.go:71]: this command has some funky logic, so just did this. Seeing if I can change it

File: pkg/ctl/cmdutils/cmd.go [hunk 4]
```diff
@@ -8,6 +8,8 @@ import (
 	"github.com/weaveworks/eksctl/pkg/eks"
 )
 
+var Silent = false
```
Comment (Skarlso) [inline pkg/ctl/cmdutils/cmd.go:11]: FWIW `var Silent bool` does result in false. :) But I understand that you wanted to be explicit. :)

Comment (Skarlso) [reply pkg/ctl/cmdutils/cmd.go:11]: I'm not liking this global variable. :/

Comment (aclevername) [reply pkg/ctl/cmdutils/cmd.go:11]: nor do I, but its the cleanest way to handle the edge-case scenarios where `NewCtl` is called multiple times. I could change the func/have an arg but am I really going to change the 30+ places where this is called for the 1 edge case? I think this is about picking the lesser evils tbh

Comment (Skarlso) [reply pkg/ctl/cmdutils/cmd.go:11]: The part that's happening more than once, can be used with a `sync.Once`. 

```go
    once.Do(func() {
        logRegionAndVersionInfo(c.ClusterConfig.Metadata)
    })
```

Or whatever it is you are guarding against here. This is global, and called only once, no matter how many times you call NewCtl.

File: pkg/ctl/cmdutils/cmd.go [hunk 5]
```diff
@@ -1,13 +1,17 @@
...
+	"sync"
+
 	"github.com/kris-nova/logger"
 	"github.com/spf13/cobra"
 
 	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
 	"github.com/weaveworks/eksctl/pkg/eks"
 )
 
+var once = &sync.Once{}
```
Comment (Skarlso) [inline pkg/ctl/cmdutils/cmd.go:13]: hmmm, I believe this should be 
```go
var once sync.Once
```

And not a shared instance 🤔

Comment (Skarlso) [reply pkg/ctl/cmdutils/cmd.go:13]: Yep, because:
> A Once must not be copied after first use.

Comment (Skarlso) [inline pkg/ctl/cmdutils/cmd.go:13]:
```suggestion
var once sync.Once
```

**Review by Skarlso — APPROVED** [review]
One change than LGTM! :)

**Review by cPu1 — APPROVED** [review]

**Pull request merged**